### PR TITLE
feat: fix custom headers get removed with @apollo/gateway@2.0.2

### PR DESCRIPTION
**Pull request by douglas-wang**

### CODE REVIEW

**Review by ftosolini — APPROVED** [review]

**Review by CCheminaud — APPROVED** [review]

**Review by 0112358132134-my — APPROVED** [review]

File: lib/FileUploadDataSource.ts [hunk 1]
```diff
@@ -216,6 +216,7 @@ export default class FileUploadDataSource extends RemoteGraphQLDataSource {
       ...request.http,
       // Apollo types are not up-to-date, make TS happy
       body: form as unknown as string,
+      headers: Object.fromEntries(request.http.headers),
```
Comment (barbieri) [inline lib/FileUploadDataSource.ts:219]: weird, if you look above in the code (line 199) you have:

```ts
    const headers = (request.http && request.http.headers) || new Headers();

    Object.entries(form.getHeaders() || {}).forEach(([k, value]) => {
      headers.set(k, value);
    });

    request.http = {
      headers,
      // ... more code
    };

    // more code ...

    const options = {
      ...request.http,
      // ... more code
    };
```

that is, there is already a `headers` that is a superset of `request.http.headers` enhanced with `form.getHeaders()` contents... thus I wonder what is this line solving?

Comment (0112358132134-my) [reply lib/FileUploadDataSource.ts:219]: For us it solve our issue. perhaps take a look here https://github.com/profusion/apollo-federation-file-upload/issues/52

Comment (barbieri) [reply lib/FileUploadDataSource.ts:219]: ok, but could you investigate further? Maybe it's the datatype (ex: it was a `Headers`, in your code it's a plain `Object`)

Comment (barbieri) [reply lib/FileUploadDataSource.ts:219]: for instance:

```ts
console.log('headers:', headers);
console.log('request.http:', request.http);
console.log('options:', options);
```

and see the differences from `headers` (variable) and `Object.fromEntries(request.http.headers)`

Comment (Tsymalyi) [reply lib/FileUploadDataSource.ts:219]: @barbieri Hello! [here i did some investigation about problem](https://github.com/profusion/apollo-federation-file-upload/issues/52#issuecomment-1144532765) 
And this PR fix problem.

Even Apollo team in RemoteGraphQLDataSource.ts [do the same](https://github.com/apollographql/federation/blob/d900859c766645cd617eb865864d326c0431a380/gateway-js/src/datasources/RemoteGraphQLDataSource.ts#L194) before send headers to **this.fetch** function

**Review by DDDKnightmare — APPROVED** [review]

File: lib/FileUploadDataSource.ts (hunk 1 above)
Comment (barbieri) [reply lib/FileUploadDataSource.ts:219]: ok, let's merge it then!

**Comment by glasser** [comment]
Thanks for fixing this! Are you planning a release with this change?

**Pull request merged**